hw_9/task2.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+const characters = [
+    { 'name': 'Barney', 'age': 36 },
+    { 'name': 'Fred', 'age': 40 },
+    { 'name': 'Jack', 'age': 50 }
+  ];
+
+// 1. Напишите функцию addCharacter(character) позволяющую добавить новый объект в массив characters. 
+//     Объект должен иметь поля name (string) и age (number)
+
+function addCharacter(character){
+    if (!('name' in character)) throw new Error (`Object ${JSON.stringify(character)} must have key 'name'`);
+    if (!('age' in character)) throw new Error (`Object ${JSON.stringify(character)} must have key 'age'`);
+    const {name, age} = character;
+    if (typeof name !== 'string') throw new Error (`Value of name ${JSON.stringify(name)} must be string`);
+    if (typeof age !== 'number') throw new Error (`Value of age ${JSON.stringify(age)} must be number`);
+
+    characters.push({name, age});
+    console.log(`Successfully added ${JSON.stringify({name, age})}`);
+    console.log(`Array is ${JSON.stringify(characters)}`);
+}
+
+addCharacter({ 'name': 'John', 'age': 25 });
+
+// 2. Напишите функцию getCharacter(name), позволяющую получить объект персонажа по его имени// getCharacter('Fred') => { 'name': 'Fred', 'age': 40 }
+
+function getCharacter(name){
+    if (typeof name !== 'string') throw new Error (`Input ${name} must be 'string'`);
+
+    const character = characters.find(character => character.name === name);
+    return character ? character : null;
+}
+
+console.log(getCharacter('Fred'));
+
+// 3. Напишите функцию getCharactersByAge(minAge), возвращающую массив персонажей НЕ МЛАДШЕ minAge // getCharactersByAge(40) => [{ 'name': 'Fred', 'age': 40 },{ 'name': 'Jack', 'age': 50 }]
+
+function getCharactersByAge(minAge) {
+    if (typeof minAge !== 'number') throw new Error (`Input ${minAge} must be 'number'`);
+
+    const minAgeValidCharacters = characters.filter(character => character.age >= minAge);
+    return JSON.stringify(minAgeValidCharacters);
```

Comment:
```suggestion
    return minAgeValidCharacters;
```
Зачем переводить в строку, если ожидается что функция вернет массив?